internal/impl/update.go

```diff
@@ -5,37 +5,48 @@ import (
 	"fmt"
 
 	"go.jetpack.io/devbox/internal/lock"
+	"go.jetpack.io/devbox/internal/nix"
 	"go.jetpack.io/devbox/internal/searcher"
 	"go.jetpack.io/devbox/internal/ux"
 	"go.jetpack.io/devbox/internal/wrapnix"
 )
 
 func (d *Devbox) Update(ctx context.Context, pkgs ...string) error {
-	var pkgsToUpdate []string
-	for _, pkg := range pkgs {
-		found, err := d.findPackageByName(pkg)
-		if err != nil {
-			return err
-		}
-		pkgsToUpdate = append(pkgsToUpdate, found)
+	inputs, err := d.inputsToUpdate(pkgs...)
+	if err != nil {
+		return err
 	}
-	if len(pkgsToUpdate) == 0 {
-		pkgsToUpdate = d.Packages()
+
+	for _, pkg := range inputs {
+		if pkg.IsLegacy() {
+			fmt.Fprintf(d.writer, "Updating %s -> %s\n", pkg.Raw, pkg.LegacyToVersioned())
+			if err := d.Remove(ctx, pkg.Raw); err != nil {
+				return err
+			}
+			if err := d.lockfile.ResolveToCurrentNixpkgCommitHash(
+				pkg.LegacyToVersioned(),
+			); err != nil {
+				return err
+			}
+			if err := d.Add(ctx, pkg.LegacyToVersioned()); err != nil {
+				return err
+			}
+		}
 	}
 
-	for _, pkg := range pkgsToUpdate {
-		if !lock.IsVersionedPackage(pkg) {
+	for _, pkg := range inputs {
+		if !lock.IsVersionedPackage(pkg.Raw) {
 			fmt.Fprintf(d.writer, "Skipping %s because it is not a versioned package\n", pkg)
 			continue
 		}
-		existing := d.lockfile.Packages[pkg]
-		newEntry, err := searcher.Client().Resolve(pkg)
+		existing := d.lockfile.Packages[pkg.Raw]
+		newEntry, err := searcher.Client().Resolve(pkg.Raw)
```

Comment:
is this related to your comment in the other PR about using input.Raw instead of input.String()?